text/0418-deprecate-route-render-methods.md

```diff
@@ -0,0 +1,249 @@
+    if (this.user.isLoggedIn) {
+      this.render('account', {
+        into: 'applcation',
+        outlet: 'account',
+        controller: 'account'
+      });
+    } else {
+      this.transitionTo('login')
+    }
+  }
+});
+```
+
+One way this could be migrated is like the following:
+
+```hbs
+{{! app/templates/authenticated.hbs}}
+
+<nav>
+  <h1>ACME Corp.</h1>
+  <section id="account-placeholder"></section>
+</nav>
+
+<section id="content">
+  {{outlet}}
+</section>
+```
+
+```hbs
+{{! app/templates/authenticated/campaigns.hbs }}
+
+{{outlet}}
+
+{{#in-element this.accountPlaceholder}}
+  {{#link-to 'account'}}
+    <img src="{{this.account.img}}" alt="{{this.account.name}} />
+  {{/link-to}}
+{{/in-element}}
+```
+
+```js
+// app/routes/authenticated.js
+import Route from '@ember/route';
+import { inject as service } from '@ember/service';
+
+export default Route.extend({
+  //...
+  user: service('user'),
+  beforeModel() {
+    if (!this.user.isLoggedIn) {
+      this.transitionTo('login')
+    }
+  }
+});
+```
+
+```js
+// app/controller/authenticated/campaigns.js
+import Route from '@ember/route';
+import Controller from '@ember/controller';
+import { inject as service } from '@ember/service';
+import { inject as controller } from '@ember/controller';
+
+export default Controller.extend({
+  //...
+  account: controller('account')
+  init() {
+    this._super(...arguments);
+    this.accountPlaceholder = document.getElementById('account-placeholder');
+  }
+});
+```
+
+If you want to do this with components you could do the same thing as the following:
+
+```hbs
+{{! app/templates/authenticated.hbs}}
+
+<nav>
+  <h1>ACME Corp.</h1>
+  <section id="account-placeholder"></section>
+</nav>
+
+<section id="content">
+  {{outlet}}
+</section>
+```
+
+```hbs
+{{! app/templates/authenticated/campaigns.hbs }}
+{{outlet}}
+
+<UserAccount />
+```
+
+```hbs
+{{! app/templates/components/user-account.hbs }}
+{{#in-element this.accountPlaceholder}}
+  {{#link-to 'account'}}
+    <img src="{{this.account.img}}" alt="{{this.account.name}} />
+  {{/link-to}}
+{{/in-element}}
+```
+
+```js
+// app/routes/authenticated.js
+import Route from '@ember/route';
+import { inject as service } from '@ember/service';
+
+export default Route.extend({
+  //...
+  user: service('user'),
+  beforeModel() {
+    if (!this.user.isLoggedIn) {
+      this.transitionTo('login')
+    }
+  }
+});
+```
+
+```js
+// app/components/user-account.js
+import Component from '@ember/route';
+import { inject as controller } from '@ember/controller';
+
+export default Component.extend({
+  // ...
+  account: controller('account'),
+  init() {
+    this._super(...arguments);
+    this.accountPlaceholder = document.getElementById('account-placeholder');
+  }
+});
+```
+
+# How We Teach This
+
+This has not been a mainline API for quite some time now. The guides do not mention this functionality at all. It is likely that the majority of Ember applications do not use these APIs.
```

Comment:
> This has not been a mainline API for quite some time now.

I think this was supposed to read "This API has not existed for very long"?

Reply:
I'm referring to the fact that `renderTemplate` and the associated APIs do not and have not played a predominate role in the programming model for quite some time now.

Reply:
This API existed in Ember 1.0, so it definitely "has existed for very long". :)